### PR TITLE
fix(input): single-line hints overflowing the parent

### LINKED ISSUE #4051: MdHint causes MdTab to show scrollbar
#### Bug, feature request, or proposal:
Bug: MdTab shows the vertical scroll bar when it contains an md-hint inside input-container.

#### What is the expected behavior?
MdTab not showing the vertical scroll bar when using md-hint inside

#### What is the current behavior?
It's showing vertical scrollbar when I use md-hint in input-container

#### What are the steps to reproduce?
Try to put and remove the md-hint of this plunker: https://plnkr.co/edit/3jAsCcMtnDr3Sh39LMap?p=preview

#### What is the use-case or motivation for changing an existing behavior?
Use md-hint

#### Which versions of Angular, Material, OS, browsers are affected?
@angular/animations: 4.0.1
@angular/cli: 1.0.0
@angular/common: 4.0.1
@angular/compiler: 4.0.1
@angular/compiler-cli: 4.0.1
@angular/core: 4.0.1
@angular/flex-layout: 2.0.0-beta.7
@angular/forms: 4.0.1
@angular/http: 4.0.1
@angular/material: 2.0.0-beta.3
@angular/platform-browser: 4.0.1
@angular/platform-browser-dynamic: 4.0.1
@angular/platform-server: 4.0.1
@angular/router: 4.0.1

#### Is there anything else we should know?
As a workaround I'm using in styles.css

.mat-tab-body {
    overflow-y: hidden !important;
}

but not sure about side efects of doing it.

### PR DESCRIPTION
**Pull request by crisbeto**
Fixes the input container hints causing an overflow on the parent by a couple of pixels. Note that this can be fixed alternatively by setting `line-height: 1` on the wrapper, however it causes multi-line hints to be a little too compact. I went with 1.2 since it seems to be the closest to the browser defaults for `normal`.

Fixes #4051.

### CODE REVIEW

File: src/lib/input/input-container.scss [hunk 1]
```diff
@@ -227,11 +226,14 @@ $mat-input-underline-disabled-background-image:
...
 .mat-input-subscript-wrapper {
+  $line-height: 1.2em;
+
   position: absolute;
   font-size: 75%;
   top: 100%;
   width: 100%;
-  margin-top: -$mat-input-wrapper-spacing;
+  margin-top: -$line-height;
+  line-height: $line-height;
```
Comment (mmalerba) [inline src/lib/input/input-container.scss]: why not change the `line-height` on `.mat-input-container` from `normal` to `1.2em` instead of overriding here?

File: src/lib/input/input-container.scss [hunk 2]
```diff
@@ -43,7 +42,7 @@ $mat-input-underline-disabled-background-image:
 // Global wrapper. We need to apply margin to the element for spacing, but
 // cannot apply it to the host element directly.
 .mat-input-wrapper {
-  margin: $mat-input-wrapper-spacing 0;
+  margin: 1em 0;
```
Comment (mmalerba) [inline src/lib/input/input-container.scss]: why did you remove the constant?

Comment (crisbeto) [inline src/lib/input/input-container.scss]: After the changes I did below, it was only being used in one place which makes it pointless.

File: src/lib/input/input-container.scss (hunk 1 above)
Comment (crisbeto) [inline src/lib/input/input-container.scss]: I wanted to avoid breaking other parts of the input container since these changes are only relevant to the hints.

File: src/lib/input/input-container.scss (hunk 2 above)
Comment (mmalerba) [inline src/lib/input/input-container.scss]: I think we often prefer constants even if they're only used once since they describe what the value is supposed to represent.

File: src/lib/input/input-container.scss (hunk 1 above)
Comment (mmalerba) [inline src/lib/input/input-container.scss]: I think it should be ok, most browsers have normal = 1.2. Using 1.2 explicitly will make sure its consistent in all browsers

**Comment by mmalerba** [comment]
LGTM

File: src/lib/input/input-container.scss (hunk 1 above)
Comment (mmalerba) [inline src/lib/input/input-container.scss]: Since changing the one on the container causes other elements to overflow I'm fine with doing this
